src/platform/with_wgl/utils.rs

```diff
@@ -0,0 +1,711 @@
+                                  as winapi::UINT, &mut output) == 0
+    {
+        return Err(format!("DescribePixelFormat function failed: {}",io::Error::last_os_error()));
+    }
+
+    if gdi32::SetPixelFormat(hdc, id, &output) == 0 {
+        return Err(format!("SetPixelFormat function failed: {}",
+                           io::Error::last_os_error()));
+    }
+
+    Ok(())
+}
+
+// Loads the WGL functions that are not guaranteed to be supported.
+//
+// The `window` must be passed because the driver can vary depending on the window's
+// characteristics.
+unsafe fn load_extra_functions(window: winapi::HWND) -> Result<wgl_ext::Wgl, String> {
+    let (ex_style, style) = (winapi::WS_EX_APPWINDOW,
+                             winapi::WS_POPUP | winapi::WS_CLIPSIBLINGS | winapi::WS_CLIPCHILDREN);
+
+    // creating a dummy invisible window
+    let dummy_window = {
+        // getting the rect of the real window
+        let rect = {
+            let mut placement: winapi::WINDOWPLACEMENT = mem::zeroed();
+            placement.length = mem::size_of::<winapi::WINDOWPLACEMENT>() as winapi::UINT;
+            if user32::GetWindowPlacement(window, &mut placement) == 0 {
+                panic!();
+            }
+            placement.rcNormalPosition
+        };
+
+        // getting the class name of the real window
+        let mut class_name = [0u16; 128];
+        if user32::GetClassNameW(window, class_name.as_mut_ptr(), 128) == 0 {
+            return Err(format!("GetClassNameW function failed: {}",
+                               io::Error::last_os_error()));
+        }
+
+        // this dummy window should match the real one enough to get the same OpenGL driver
+        let title = OsStr::new("Dummy")
+            .encode_wide()
+            .chain(Some(0).into_iter())
+            .collect::<Vec<_>>();
+        let win = user32::CreateWindowExW(ex_style,
+                                          class_name.as_ptr(),
+                                          title.as_ptr(),
+                                          style,
+                                          winapi::CW_USEDEFAULT,
+                                          winapi::CW_USEDEFAULT,
+                                          rect.right - rect.left,
+                                          rect.bottom - rect.top,
+                                          ptr::null_mut(),
+                                          ptr::null_mut(),
+                                          kernel32::GetModuleHandleW(ptr::null()),
+                                          ptr::null_mut());
+
+        if win.is_null() {
+            return Err(format!("CreateWindowEx function failed: {}",
+                               io::Error::last_os_error()));
+        }
+
+        let hdc = user32::GetDC(win);
+        if hdc.is_null() {
+            let err = Err(format!("GetDC function failed: {}", io::Error::last_os_error()));
```

Comment:
nit: `return Err(..)`. Also, it seems you need to clean up the window you just created.